spec/models/regulator_spec.rb

```diff
@@ -92,4 +106,53 @@
       end
     end
   end
+
+  describe "#batch_inspect!" do
+    context "with a few records" do
+      before do
+        ActiveRecord::Base.transaction do
+          10.times { create(:reading, :day_time, {temp: 75, outdoor_temp: 50 }) }
+        end
+      end
+
+      it "updates violation status in batches", :vcr do
+        expect(Reading.where(violation: true)).to have(0).items
+
+        Reading.order(created_at: :asc).first.update(temp: 65)
+        Reading.order(created_at: :asc).last.update(temp: 65)
+
+        Regulator.new(Reading.all).batch_inspect!(silent: true)
+        expect(Reading.where(violation: true)).to have(2).items
+      end
+    end
+
+    context "with many records" do
```

Comment:
In prod, we may want to do this on 100,000+ records. I wanted to do a trial run with a larger number of records locally first, but didn't want that test as one of our trunk tests that we run every time we ship, hence the xit and comment. I considered deleting it but don't want to have to start from scratch if I want to do it again.